src/App.js

```diff
@@ -1,16 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatMessagesData, setChatMessagesData] = useState(chatMessages);
+  const users = [];
+  let likesCount = 0;
+  let user1 = '';
+  let user2 = '';
+
+  const updateChatData = (updatedMessage) => {
+    const messages = chatMessagesData.map((message) => {
+      if (message.id === updatedMessage.id) {
+        return updatedMessage;
+      } else {
+        return message;
+      }
+    });
+
+    setChatMessagesData(messages);
+  };
+
+  for (let i = 0; i < chatMessagesData.length; i++) {
+    let message = chatMessagesData[i];
+    if (users.indexOf(message['sender']) === -1) {
+      users.push(message['sender'])
+    }
+    if (message['liked']) {
+      likesCount++;
+    }
+  }
+
+  if (users.length === 2) {
+    user1 = users[0].toUpperCase();
+    user2 = users[1].toUpperCase();
+  }
```

Comment:
You could also directly hard code these user names into your `h1` element and remove this logic here so line 44 would look like:
```javascript 
<h1>CHAT BETWEEN {{chatMessages[0].sender}} AND {{chatMessages[1].sender}} </h1>
```